src/main/java/com/amalitech/gpuconfigurator/exception/CustomExceptionHandler.java

```diff
@@ -23,14 +25,18 @@ public ResponseEntity<Object> handleBadCredentialException(Exception e) {
         return buildResponse(HttpStatus.UNAUTHORIZED, e.getMessage());
     }
 
-    @ExceptionHandler(UsernameNotFoundException.class)
-    public ResponseEntity<Object> usernameNotFoundException(Exception e) {
-        return buildResponse(HttpStatus.NOT_FOUND, e.getMessage());
+    @ExceptionHandler({ UsernameNotFoundException.class, EntityNotFoundException.class })
```

Comment:
Handle UserNameNotFoundException Separately. It should not have a It should have 401 status instead